app/routes/_layout+/assets.$assetId.tsx

```diff
@@ -136,6 +140,11 @@ export const links: LinksFunction = () => [
 ];
 
 export default function AssetDetailsPage() {
+  const queryLocation = useLocation();
+  const queryParams = new URLSearchParams(queryLocation.search);
```

Comment:
This works but I would like to refactor it to do it the 'proper' Remix way. Please check this: https://remix.run/docs/en/main/hooks/use-search-params

So you can do:
```ts

const [searchParams] = useSearchParams()
const isQr = searchParams.get("ref") === "qr"
```